---
layout: default
title: Single-cell RNA-seq data analysis with R
description: an introductory course organized by the Princess Maxima Single Cell facility
---

<!-- The source of this page is

https://github.com/scgenomics/introScRNAseqDataAnalysis.github.io 

(docs/index.md)

which is automatically rendered as

https://scgenomics.github.io/introScRNAseqDataAnalysis.github.io/

after pushing to github. For problems with rendering  check 

https://github.com/scgenomics/introScRNAseqDataAnalysis.github.io/deployments/activity_log?environment=github-pages

-->

![logo](img/scgenomics.png)
<!-- for {{ site.variables }} see _config.yml -->
# Single cell RNA-seq data analysis with R

### [Schedule](schedule.md)

## Dates

This two-day hands-on course will be held **{{ site.course_date }}**,
from 9:30 - 12:30 and  from 13:30 - 16:30.

## Location

The Princess Maxima Foyer Auditorium.

## Intended audience

People that do in-depth analyses of single-cell RNA-seq data using
R. (Familiarity with R is a prerequisite, see below).

## Registration

You can register [through the Princess Maxima Academy]({{site.registration_url}}), or using this QR-code:<br>
<img src="img/registration-QR.png" alt="QR-code" style="width:80px;" /><br>
Having succesfully completed the course, participants are awarded 0.5
ECTS.

**Please note:** your registration implies an actual commitment to
participate the two full days of the course. Since we have limited
capacity, your absence would deny someone elseâ€™s participation. Failing
to show up without a valid reason will incur costs.


## Computing environment

We will work with [RStudio](https://www.rstudio.com/) which provides the
[R statistical programming
environment](https://www.r-project.org/). There is no need to install
RStudio yourself (except perhaps for the
[self-assessment](self-assess/R-selfassess.md), see below).

## Hardware

For this course we will provide you with software that connects to
a remote RStudio server. This remote server will contain everything
you need for this course. **Please bring your laptop**, or let us know 
in advance and we can provide you with one. 

## Learning objectives

After this course, you should be able to:

- load the data and select good cells that can be trusted
- normalize the data appropriately
- perform decontamination if needed 
- apply simple batch effect corrections
- determine confounder genes
- find reasonable parameters for dimensionality reduction and clustering
- perform Differential Gene Expression
- do automatic cell type identification
- call copy number variations

## Prerequisities

To participate in this hands-on course you should be reasonably fluent
in using R.  We will be using R and various R packages, including
[Seurat](https://satijalab.org/seurat/). If you are not quite sure if
your R skills are up to scratch, please do the following self-assement.

### Self-assessment

[This exercise](self-assess/R-selfassess.md) should take you no more than around 15
minutes.

## Lecturers

- Tito Candelli
- Lindy Visser
<!-- - Wout Megchelenbrink --> 
- Philip Lijnzaad
- Thanasis Margaritis

## Additional information

- <mailto:scgenomics@princessmaximacentrum.nl> for substantional and technical matters
- <mailto:academy@prinsesmaximacentrum.nl> for practical matters
